doc/tutorials/ostro-oic-tutorial.md

```diff
@@ -0,0 +1,1030 @@
+```
+
+### Showing a nice error message to user
+
+If we run our `controller-oic.fbp` as is, it will end up showing an error
+message like:
+
+```
+$ sol-fbp-runner controller-oic.fbp
+
+WRN:sol-flow ./src/lib/flow/sol-flow-static.c:341 flow_send_do() Error packet '22 (Argument position (1) is greater than arguments length (1))' sent from 'controller-oic.fbp (0x560d1c548e90)' was not handled
+```
+
+This is so because we didn't pass the device id as argument. But how a
+user would know? Every Soletta node type as an implicit `ERROR` port,
+where error packets are sent. The idea will be capture an ERROR packet
+sent by `cmdline_arguments` and then print a more meaningful message to
+user:
+
+```
+cmdline_arguments ERROR -> IN cmdline_usage(converter/empty-to-string:output_value="Missing device ID in the command line. Example: 2f3089a1dbfb43d38cab64383bdf9380") OUT -> IN _(console:prefix="ERROR: ")
+cmdline_usage OUT -> IN _(converter/empty-to-int:output_value=1) OUT -> CODE _(app/quit)
+```
+
+See that we used an `app/quit` node type to quit application with an
+error code, in this case, 1.
+
+Now, if we run without the expected argument, we get a better error
+message:
+
+```
+$ sol-fbp-runner controller-oic.fbp
+
+ERROR: Missing device ID in the command line. Example: 2f3089a1dbfb43d38cab64383bdf9380 (string)
+```
+
+### Running the server
+
+To really test our OIC controller, we need to run Soletta
+`light-server.fbp` sample, using the device id we are going to pass
+to the application. To do so, you can run, on Soletta dir:
+
+```
+SOL_MACHINE_ID=2f3089a1dbfb43d38cab64383bdf9380 sol-fbp-runner src/samples/flow/oic/light-server.fbp
+```
+
+And them, run our app:
+
+```
+$ sol-fbp-runner controller-oic.fbp 2f3089a1dbfb43d38cab64383bdf9380
+Server state is: false (boolean)
+Server power level is: 76 (integer range)
+Power value changed: 76 (integer range)
+State changed: true (boolean)
+Power value changed: 77 (integer range)
+```
+
+**Note:** You may see some warnings related to OIC as well, as it tries
+to get information on all your network interfaces and fail in some
+of them. They are harmless.
+
+## Step 5
+
+### Defining our own HTTP server
+
+If instead of connect to an OIC server, we connect to our own HTTP light
+server? Yeah, it can be done!
+We will first write our server and them a controller for it.
+Again, we'll use subflows for this, so we keep this code separated.
+Soletta has node types to HTTP servers, we are going to use two of them:
+`http-server/int` and `http-server/boolean`, to provide our 'power' and
+'state' values. On a `http-light-server.fbp` write:
+
+```
+power(http-server/int:path="/power",port=8002)
+state(http-server/boolean:path="/state",port=8002)
+
+_(constant/string:value="http-light-server running on port 8002, paths: /state and /power") OUT -> IN _(console:prefix="note: ")
+```
+
+First two lines create two HTTP servers, one for 'power' and another for
```

Comment:
clarify soletta will create an internal server, only one per port and then register paths as needed. Users didn't understand such properties and were asking themselves if Apache or what was being started... expecting bugs due 2 servers being used.

They were also expecting the FBP to halt in "loop" when server was created... pretty much like Python or JS servers where you have a trailing "run-forever" at the bottom. Then explain these will not block.